models/aeif_cbvg_2010.cpp

```diff
@@ -0,0 +1,657 @@
+#include <iostream>
+#include <limits>
+
+// Includes from libnestutil:
+#include "numerics.h"
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "nest_names.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+#include "integerdatum.h"
+
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+nest::RecordablesMap< nest::aeif_cbvg_2010 >
+  nest::aeif_cbvg_2010::recordablesMap_;
+
+namespace nest
+{
+/*
+ * template specialization must be placed in namespace
+ *
+ * Override the create() method with one call to RecordablesMap::insert_()
+ * for each quantity to be recorded.
+ */
+template <>
+void
+RecordablesMap< aeif_cbvg_2010 >::create()
+{
+  // use standard names whereever you can for consistency!
+  insert_(
+    names::V_m, &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::V_M > );
+  insert_(
+    names::w, &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::W > );
+  insert_(
+    names::z, &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::Z > );
+  insert_(
+    names::V_T, &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::V_T > );
+  insert_( names::u_bar_plus,
+    &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::U_BAR_PLUS > );
+  insert_( names::u_bar_minus,
+    &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::U_BAR_MINUS > );
+  insert_( names::u_bar_bar,
+    &aeif_cbvg_2010::get_y_elem_< aeif_cbvg_2010::State_::U_BAR_BAR > );
+}
+}
+
+
+extern "C" int
+nest::aeif_cbvg_2010_dynamics( double,
+  const double y[],
+  double f[],
+  void* pnode )
+{
+  // a shorthand
+  typedef nest::aeif_cbvg_2010::State_ S;
+
+  // get access to node so we can almost work as in a member function
+  assert( pnode );
+  const nest::aeif_cbvg_2010& node =
+    *( reinterpret_cast< nest::aeif_cbvg_2010* >( pnode ) );
+
+  const bool is_refractory = node.S_.r_ > 0;
+  const bool is_clamped = node.S_.clamp_r_ > 0;
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  // The following code is verbose for the sake of clarity. We assume that a
+  // good compiler will optimize the verbosity away...
+
+  // Clamp membrane potential to V_reset while refractory, otherwise bound
+  // it to V_peak.
+  const double& V = ( is_refractory || is_clamped )
+    ? ( is_clamped ? node.P_.V_clamp_ : node.P_.V_reset_ )
+    : std::min( y[ S::V_M ], node.P_.V_peak_ );
+  // shorthand for the other state variables
+  const double& w = y[ S::W ];
+  const double& z = y[ S::Z ];
+  const double& V_T = y[ S::V_T ];
+  const double& u_bar_plus = y[ S::U_BAR_PLUS ];
+  const double& u_bar_minus = y[ S::U_BAR_MINUS ];
+  const double& u_bar_bar = y[ S::U_BAR_BAR ];
+
+  const double I_spike = ( node.P_.g_L * node.P_.Delta_T
```

Comment:
```suggestion
  const double I_spike = (node.P_.Delta_T == 0.) ? 0. : ( node.P_.g_L * node.P_.Delta_T
```

Reply:
Good catch, I missed that when I reconstructed the case `Delta_T = 0` from `aeif_psc_delta`! Fixed with 40ea525.